vm/ieleClientUtils.ml

```diff
@@ -148,18 +132,13 @@ let send_request ctx =
   let addr = Unix.ADDR_INET(Unix.inet_addr_loopback,(int_of_string Sys.argv.(2))) in
   World.send addr ctx
 
-let test_transaction header (state: (string * Basic.json) list) (tx: Basic.json) (result: Basic.json) : (string * Basic.json) list =
+let exec_transaction header (state: (string * Basic.json) list) (tx: Basic.json) : (string * Basic.json) list * call_result =
   let gas_price = of_hex (tx |> member "gasPrice" |> to_string) in
   let gas_provided = of_hex (tx |> member "gasLimit" |> to_string) in
   let owner = tx |> member "to" |> to_string in
   let txcreate = owner = "" in
-  let secretkey = of_hex_unsigned (tx |> member "secretKey" |> to_string) in
-  let secretkey_buffer = Bigarray.Array1.of_array Bigarray.char Bigarray.c_layout (Array.init 32 (fun idx -> Bytes.get secretkey idx)) in
-  let pubkey = Secp256k1.Public.of_secret context (Secp256k1.Secret.read_exn context secretkey_buffer) in
-  let pubkey_string = String.init 64 (fun idx -> Bigarray.Array1.get (Secp256k1.Public.to_bytes ~compress:false context pubkey) (idx+1)) in
-  let pubkey_hash = Cryptokit.hash_string (hash()) pubkey_string in
-  let origin = String.sub pubkey_hash 12 20 in
-  let origin = Bytes.of_string origin in
+  let from = tx |> member "from" |> to_string in
```

Comment:
What happened to this secret key handling?
I don't see anything like this in `ieleVmTest.ml` either.

Reply:
none of our code paths (solidity-to-iele, mantis, etc) relied on having signed transactions, so seeing as the signed transactions introduced complexity and additional dependencies into IELE, I decided to simplify the test format and just remove the transaction signatures. This will be part of an upcoming iele-semantics PR that will be ready after this PR is merged.